lib/modules/manager/pep621/schema.ts

```diff
@@ -8,6 +8,46 @@ const DependencyRecordSchema = z
   .record(z.string(), z.array(z.string()))
   .optional();
 
+const PdmSchema = z.object({
```

Comment:
I would like to see this refactor in a separate PR. 🤗

Reply:
Sure thing, thought it was small enough to not require a separate PR, but since 2 PRs depend on it, might be better anyway :)
https://github.com/renovatebot/renovate/pull/31280